src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -2326,10 +2326,15 @@ void C2_MacroAssembler::expand_bits_l_v(Register dst, Register src, Register mas
 }
 
 void C2_MacroAssembler::element_compare(Register a1, Register a2, Register result, Register cnt, Register tmp1, Register tmp2,
-                                        VectorRegister vr1, VectorRegister vr2, VectorRegister vrs, bool islatin, Label &DONE,
-                                        Assembler::LMUL lmul) {
+                                        VectorRegisterGroup vg1, VectorRegisterGroup vg2, VectorRegisterGroup vgs, bool islatin, Label &DONE) {
   Label loop;
   Assembler::SEW sew = islatin ? Assembler::e8 : Assembler::e16;
+  assert(vgrp_to_lmul(vg1) == vgrp_to_lmul(vg2), "sanity");
+  assert(vgrp_to_lmul(vg1) == vgrp_to_lmul(vvgss), "sanity");
+  Assembler::LMUL lmul = vgrp_to_lmul(vg1);
+  VectorRegister vr1 = vg1.start_vreg();
```

Comment:
I would express it as `vg1.as_vreg()`. I still think we want to keep an explicit conversion, or that might bring some weird bugs, but it's still a conversion.